### PR TITLE
PR 3/3 - feat(gamification): expose weekly score history through existing endpoint (#300)

### PR DESCRIPTION
**Pull request by eariasvalor**
⚠️ Do not merge until PR 2/3 (#296) is merged first. This PR depends on `getUserScoresHistoryChart()` and `ScoresHistoryResponseDto` introduced there.

## Description

PR 3/3 of parent issue #281. Updates `UserScoreController` to return the new weekly aggregated score history, replaces dead DTOs and updates Swagger documentation.

This PR modifies the existing `GET /users/{userId}/scores/history` endpoint contract.

## Analysis of existing behavior

The endpoint previously returned `PointsHistoryResponseDto` — a flat list of score entries with no aggregation. `getUserPointsHistory()` in the service layer is now dead code after PR 2.

## What this PR changes

- `UserScoreController` now calls `getUserScoresHistoryChart()` and returns `ScoresHistoryResponseDto`
- Swagger/OpenAPI updated to reflect the new response schema
- `getUserPointsHistory()` and `buildHistoryResponse()` removed from `UserScoreService` and `UserScoreServiceImpl`
- `PointsHistoryResponseDto` and `PointHistoryEntryDto` deleted — dead code
- `UserScoreControllerTest` updated to use the new DTO and JSON paths
- CHANGELOG updated to `3.5.0-RELEASE`
- Unit tests added for `ScoresHistoryResponseDto` and `WeeklyPointsDto` (all passing, see the screenshot above)

test-results-controller-weekly-history-pr3

### Swagger documentation
swagger-endpoint:
swagger-endpoint
swagger-schema:
swagger-schema

### 200 — Weekly history response
postman-get-scores-history-200-weekly-response-pr3:
postman-get-scores-history-200-weekly-response-pr3 png

### 400 — Invalid UUID
postman-get-scores-history-400-invalid-uuid-pr3:
postman-get-scores-history-400-invalid-uuid-pr3 png

## Out of scope

- Filtering history by `activityType` — tracked in #311
- Configurable `aggregationType` (daily, monthly) — tracked in #310

## Technical debt

- `aggregationType` hardcoded to `"WEEKLY"` — tracked in #310
- `activityType` filtering not yet supported — tracked in #311

## Self-checklist

- [X] I tested my changes locally and verified they work as expected
- [X] The PR has a clear and focused purpose (only one feature, fix or refactor)
- [X] Title, description, and changelog (if needed) are filled in correctly
- [X] There are no leftover merge conflicts or unrelated changes from previous branches
- [X] All automated checks (like SonarCloud) have passed
- [X] I responded to all previous review comments and only resolved those I truly addressed
- [X] I ran SonarLint locally on the modified files and confirmed there are no warnings or errors

### CODE REVIEW

**Review by carlosPc1987 — CHANGES_REQUESTED** [review]
Looks good overall. Could you add one Swagger screenshot + one executed response example (Swagger or Postman) for GET /users/{userId}/scores/history with the new aggregated schema? Since this PR changes an existing endpoint contract, having that evidence in the PR would help reviewers and frontend alignment.

**Comment by eariasvalor** [comment]
@carlosPc1987 Added Postman screenshots to the PR description validating both scenarios:
- 200 with the new weekly aggregated response structure
- 400 with invalid UUID

Also added Swagger screenshots showing the updated endpoint documentation and response schema.

**Review by carlosPc1987 — APPROVED** [review]
Thanks @eariasvalor — the added Swagger + Postman screenshots in the PR description cover the new aggregated response for `GET /itachallenge/api/v1/users/{userId}/scores/history` (200 + invalid UUID), which addresses the contract-change concern.

**LGTM** from my side — I’m resolving my previous review comment.

**Review by JordiCasas87 — CHANGES_REQUESTED** [review]
Looks good overall, nice refactor !

I also noticed that some tests from the previous implementation were removed (which makes sense given the refactor). Just wondering if the new aggregation logic has equivalent coverage for key scenarios (e.g. ordering, edge cases, etc.).

Just bringing this up so we can align before approving; once we confirm whether additional coverage is needed, I’m happy to move forward.

**Comment by eariasvalor** [comment]
Good question! The aggregation logic is fully covered, but the tests live in PR 2 (#296) since that's where the service logic was introduced. Specifically:

- `UserScoreServiceImplTest` covers: weekly grouping, progressive accumulated total, ISO week format, totalPoints equals last accumulatedAtEnd, empty history, and mixed sources in the same week
- `UserScoreControllerTest` covers: 200 with weekly structure, 200 with empty history, and 400 with invalid UUID
- `ScoresHistoryResponseDtoTest` and `WeeklyPointsDtoTest` cover the new DTOs

This PR removes the old tests that referenced `PointsHistoryResponseDto` since that DTO no longer exists, and adds the DTO unit tests. The core aggregation coverage is in PR 2.

**Review by JordiCasas87 — APPROVED** [review]
Thanks for the clarification, makes sense. :)

All good from my side, approving.

**Comment by kenan-it-academy** [comment]
This PR is a bit large. Skipping for now.

File: itachallenge-challenge/src/test/java/com/itachallenge/gamification/service/UserScoreServiceImplTest.java [hunk 1]
```diff
@@ -40,115 +40,6 @@ private UserScoreDocument scoreDoc(int points, LocalDateTime date) {
                 .build();
     }
 
-    @Test
```
Comment (kenan-it-academy) [inline itachallenge-challenge/src/test/java/com/itachallenge/gamification/service/UserScoreServiceImplTest.java:43]: These tests shouldn't be getting deleted (Unless I'm missing something)

**Review by kenan-it-academy — CHANGES_REQUESTED** [review]
Careful with accidentally erasing tests through git nonsense!

**Comment by eariasvalor** [comment]
@kenan-it-academy please take a look at my reply to your issue: the tests are part of the dead code we are removing in this PR so they should be deleted.

**Review by kenan-it-academy — APPROVED** [review]
Seems reasonable!
